pkg/neonvm/controllers/vm_controller.go

```diff
@@ -1222,7 +1222,7 @@ func affinityForVirtualMachine(vm *vmv1.VirtualMachine) *corev1.Affinity {
 					{
 						Key:      "kubernetes.io/arch",
 						Operator: "In",
-						Values:   []string{"amd64"},
```

Comment:
What's the reasoning behind this change,. why does VM's affinity must match controller's architecture?

If we might have heterogeneous clusters, maybe we need to specify the architecture explicitly in the spec?

Reply:
We already can specify it through the VM spec, by passing affinity. 

I think at the moment using by default the same arch as the controller might be a safe assumption. Do you have any other options in your mind?

Reply:
> Maybe it worth to discuss whether we want to have heterogeneous clusters.

The existence of https://github.com/neondatabase/autoscaling/pull/1123 hints toward yes. In that case, we need to align on how it should be determined where to schedule the VM.

Reply:
@Omrigan do you have any suggestions?

Reply:
we agreed verbally with @sharnoff to keep it as current PR has, but change PR #1083 with explicit spec.arch field